pyked/converters.py

```diff
@@ -0,0 +1,741 @@
+
+    # Get properties shared across the file
+    properties['common-properties'] = get_common_properties(root)
+
+    # Determine definition of ignition delay
+    properties['common-properties']['ignition-type'] = get_ignition_type(root)
+
+    # Now parse ignition delay datapoints
+    properties['datapoints'] = get_datapoints(root)
+
+    # Get compression time for RCM, if volume history given
+    # if 'volume' in properties and 'compression-time' not in properties:
+    #     min_volume_idx = numpy.argmin(properties['volume'])
+    #     min_volume_time = properties['time'][min_volume_idx]
+    #     properties['compression-time'] = min_volume_time
+
+    # Ensure combinations of volume, time, pressure-rise are correct.
+    if ('volume' in properties['common-properties'] and
+        'time' not in properties['common-properties']
+        ):
+        raise KeywordError('Time values needed for volume history')
+    elif (any(['volume' in dp for dp in properties['datapoints']]) and
+          not any(['time' in dp for dp in properties['datapoints']])
+          ):
+        raise KeywordError('Time values needed for volume history')
+
+    if ('compression-time' in properties['common-properties'] or
+        any([dp for dp in properties['datapoints'] if dp.get('compression-time')])
+        ) and properties['apparatus']['kind'] == 'shock tube':
+        raise KeyError('Compression time cannot be defined for shock tube.')
+
+    if ('pressure-rise' in properties['common-properties'] or
+        any([dp for dp in properties['datapoints'] if dp.get('pressure-rise')])
+        ) and properties['apparatus']['kind'] == 'rapid compression machine':
+        raise KeyError('Pressure rise cannot be defined for RCM.')
+
+    if (('volume' in properties['common-properties'] and
+         'pressure-rise' in properties['common-properties']
+         ) or ('volume' in properties['common-properties'] and
+               any([dp for dp in properties['datapoints'] if dp.get('pressure-rise')])
+               ) or ('pressure-rise' in properties['common-properties'] and
+                     any([dp for dp in properties['datapoints'] if dp.get('volume')])
+                     )
+        ):
+        raise KeywordError('Both volume history and pressure rise '
+                           'cannot be specified'
+                           )
+
+    # apply any overrides
+    if file_author:
+        properties['file-author']['name'] = file_author
+    if file_author_orcid:
+        properties['file-author']['ORCID'] = file_author_orcid
+
+    # Now go through datapoints and apply common properties
+    for idx in range(len(properties['datapoints'])):
+        for prop in properties['common-properties']:
+            properties['datapoints'][idx][prop] = properties['common-properties'][prop]
+
+    # compression time doesn't belong in common-properties
+    properties['common-properties'].pop('compression-time', None)
```

Comment:
It turns out `compression-time` isn't supported by ReSpecTh anyways, so I'll remove all this.